### PR TITLE
test on py3.12

### PR DESCRIPTION
**Pull request by graingert**

### CODE REVIEW

**Comment by sethmlarson** [comment]
@graingert Is pytest-httpbin==2.0.0rc1 going to be promoted to stable?

**Comment by graingert** [comment]
@sethmlarson hopefully, just blowing the dust off of CI https://github.com/kevin1024/pytest-httpbin/pull/76

can you get someone to look at https://github.com/postmanlabs/httpbin/pull/684/files ?

**Comment by nateprewitt** [comment]
@graingert, httpbin appears to be in a pretty rough state. It was moved under @postmanlabs when they started hosting the live endpoint but has since been relatively abandoned. They kicked all the previous maintainers off the project and don't appear to have any resourcing allocated to it. I've tried reaching out a few times without any meaningful response.

Going forward, we may either need to fork it or move Requests onto a different testing dependency.

**Comment by nateprewitt** [comment]
I cut https://github.com/postmanlabs/httpbin/issues/702 to see if we can get a more public response there on maintenance status.

File: requirements-dev.txt [hunk 1]
```diff
@@ -1,7 +1,7 @@
 -e .[socks]
 pytest>=2.8.0,<=6.2.5
 pytest-cov
-pytest-httpbin==1.0.0
+pytest-httpbin==2.0.0rc1
```
Comment (graingert) [inline requirements-dev.txt]:
```suggestion
pytest-httpbin==2.0.0rc2
```

File: setup.py [hunk 2]
```diff
@@ -65,7 +65,7 @@ def run_tests(self):
     "certifi>=2017.4.17",
 ]
 test_requirements = [
-    "pytest-httpbin==0.0.7",
+    "pytest-httpbin==2.0.0rc1",
```
Comment (graingert) [inline setup.py]:
```suggestion
    "pytest-httpbin==2.0.0rc2",
```

File: requirements-dev.txt [hunk 3]
```diff
@@ -1,7 +1,7 @@
 -e .[socks]
 pytest>=2.8.0,<=6.2.5
 pytest-cov
-pytest-httpbin==1.0.0
+pytest-httpbin==2.0.0rc2
```
Comment (graingert) [inline requirements-dev.txt]:
```suggestion
pytest-httpbin==2.0.0
```

File: setup.py [hunk 4]
```diff
@@ -65,7 +65,7 @@ def run_tests(self):
     "certifi>=2017.4.17",
 ]
 test_requirements = [
-    "pytest-httpbin==0.0.7",
+    "pytest-httpbin==2.0.0rc2",
```
Comment (graingert) [inline setup.py]:
```suggestion
    "pytest-httpbin==2.0.0",
```

**Comment by graingert** [comment]
> @graingert Is pytest-httpbin==2.0.0rc1 going to be promoted to stable?

@sethmlarson I've just pushed v2.0.0 to pypi!

**Review by nateprewitt — APPROVED** [review]

**Comment by nateprewitt** [comment]
Thanks, @graingert!

**Comment by kevin1024** [comment]
> They kicked all the previous maintainers off the project and don't appear to have any resourcing allocated to it. I've tried reaching out a few times without any meaningful response.

Well, I seem to have been removed as a maintainer on the github repo, but I am still a maintainer on pypi. Perhaps I can help with a new release!  

I also (occasionally) maintain https://github.com/kevin1024/pytest-httpbin which relies on httpbin, and https://github.com/kevin1024/vcrpy/

Shall we fork under [psf](https://github.com/psf/) or would you like me to fork under my own name?

**Comment by nateprewitt** [comment]
Hey @kevin1024, thanks for checking in! I believe you, Ian, and myself are the three with release perms on PyPI so we can definitely go that direction. I was going to wait out a response on the issue above for a bit longer before we take on the task of forking.

I think there are also a number of changes Kenneth merge between the last release and now that may need vetting before we cut something new. If we don't hear anything back going into July, I think that's probably time to look at taking action.

Edit: Don't let me stop you from doing this sooner if you're motivated. I just wanted to minimize churn/maintenance overhead if possible.

**Comment by kevin1024** [comment]
Hello fellow maintainer!  Works for me.  Let's wait a bit longer for a response before forking.  My main motivation for forking would be to fix:

https://github.com/postmanlabs/httpbin/issues/673

So that I can fix:

https://github.com/kevin1024/pytest-httpbin/issues/72

So that I can fix:

https://github.com/kevin1024/vcrpy/issues/645

Basically: https://www.youtube.com/watch?v=AbSehcT19u0

**Pull request merged**